src/hooks/useRoom.ts

```diff
@@ -1,53 +1,31 @@
-import { useCallback, useMemo } from "react";
+import { useCallback } from "react";
 
 import { Room } from "types/rooms";
 
-import { getExternalRoomSlug } from "utils/room";
-import { enterVenue } from "utils/url";
-import { enterExternalRoom } from "utils/userLocation";
-
-import { useRelatedVenues } from "hooks/useRelatedVenues";
-import { useRecentLocationUsers } from "hooks/users";
-import { useUser } from "hooks/useUser";
+import {
+  enterVenue,
+  getLastUrlParam,
+  getUrlWithoutTrailingSlash,
+} from "utils/url";
 
 export interface UseRoomProps {
   room?: Room;
-  venueName: string;
 }
-// @debt refactor useRoom to take venueId instead of venueName
-export const useRoom = ({ room, venueName }: UseRoomProps) => {
-  const { user } = useUser();
-  const userId = user?.uid;
-
+export const useRoom = ({ room }: UseRoomProps) => {
```

Comment:
I find `useRoom` hook a bit confusing because it:

- is called `useRoom` but accepts `room` as a parameter
- returns `enterRoom` which implies some control over the room entrance
- returns `portalVenueId` which conflicts with the hook name

Maybe it's better to split this into two separate hooks: one for entering the room and one for getting room (or portal) venueId and name them accordingly? It may be out of scope of this PR but it undergoes significant changes here so might as well go even further.